ledger/metrics/src/test/suite/scala/com/daml/metrics/InstrumentedSourceSpec.scala

```diff
@@ -160,4 +239,19 @@ object InstrumentedSourceSpec {
 
   }
 
+  // For testing only, provides a sampled sequence of the state of the counter until finishSampling is called.
+  private final case class SamplingCounter(samplingMillis: Long) extends Counter {
```

Comment:
Not sure why this is a `case` class. Copying it sounds dangerous.

Reply:
I do case class all the time for the generated goodies and because `new` gives me the creeps.
if it causes confusion, that is good enough reason for me to change this :+1: